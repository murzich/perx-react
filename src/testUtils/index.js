// eslint-disable-next-line import/prefer-default-export
export { default as renderWithRedux } from './renderWithRedux';
