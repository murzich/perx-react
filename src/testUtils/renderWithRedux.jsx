// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../state/reducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export const rerenderWithRedux = rerender => (
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) => ({
  ...rerender(<Provider store={store}>{ui}</Provider>),
  store,
});
