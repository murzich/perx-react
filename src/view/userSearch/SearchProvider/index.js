import SearchProvider from './SearchProvider';

export default SearchProvider;
