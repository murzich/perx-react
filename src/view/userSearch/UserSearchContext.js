import { createContext } from 'react';

const UserSearchContext = createContext();

export default UserSearchContext;
