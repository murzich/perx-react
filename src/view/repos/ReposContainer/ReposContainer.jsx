import PropTypes from 'prop-types';
import React, { useContext, useEffect } from 'react';
import { connect } from 'react-redux';
import { reposActions, reposSelectors } from '../../../state';
import ReposList from '../ReposList';
import { UserSearchContext } from '../../userSearch';

const ReposContainer = ({ fetchReposAct, repos }) => {
  const { userName } = useContext(UserSearchContext);
  useEffect(() => {
    if (userName) {
      fetchReposAct(userName);
    }
  }, [fetchReposAct, userName]);

  return <ReposList repos={repos} />;
};

ReposContainer.propTypes = {
  fetchReposAct: PropTypes.func.isRequired,
  ...ReposList.propTypes,
};

const mapStateToProps = state => ({
  repos: reposSelectors.getCurrentReposSel(state),
});

export default connect(
  mapStateToProps,
  {
    fetchReposAct: reposActions.fetchReposRequest,
  },
)(ReposContainer);
