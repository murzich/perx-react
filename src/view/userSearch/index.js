import SearchForm from './SearchForm';
import SearchProvider from './SearchProvider';
import UserSearchContext from './UserSearchContext';

export { UserSearchContext, SearchProvider };

export default SearchForm;
