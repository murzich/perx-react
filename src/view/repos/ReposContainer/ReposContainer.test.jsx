import { cleanup } from '@testing-library/react';
import React from 'react';
import reposStub from '../../../../__stubs__/repos';
import { renderWithRedux } from '../../../testUtils';
import { UserSearchContext } from '../../userSearch';
import ReposContainer from './ReposContainer';

describe('ReposContainer', () => {
  afterEach(cleanup);

  const testUser = 'testUser';

  it('should match snapshot with default props', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <ReposContainer />
      </UserSearchContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with repos', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <ReposContainer />
      </UserSearchContext.Provider>,
      {
        initialState: {
          repos: {
            repos: {
              [reposStub.data[0].id]: reposStub.data[0],
            },
            order: [reposStub.data[0].id],
          },
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
