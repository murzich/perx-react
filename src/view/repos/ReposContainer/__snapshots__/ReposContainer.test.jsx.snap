// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposContainer should match snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="ui divided relaxed list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          Enter user name to search for his repositories
        </div>
        <div
          class="description"
        >
          There should be a list of all his repositories with public access.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReposContainer should match snapshot with repos 1`] = `
<DocumentFragment>
  <div
    class="ui divided relaxed list"
    role="list"
  >
    <div
      class="item"
      role="listitem"
    >
      <div
        class="content"
      >
        <a
          class="header"
          href="https://github.com/gaearon/6to5"
        >
          6to5
        </a>
        <div
          class="description"
        >
          Turn ES6+ code into readable vanilla ES5 with source maps and more!
        </div>
        <div
          class="ui small labels tags"
        >
          <div
            class="ui label"
          >
            <i
              aria-hidden="true"
              class="star icon"
            />
            1
          </div>
          <div
            class="ui label"
          >
            <i
              aria-hidden="true"
              class="fork icon"
            />
            1
          </div>
          <div
            class="ui label"
          >
            <i
              aria-hidden="true"
              class="eye icon"
            />
            1
          </div>
          <div
            class="ui label"
          >
            JavaScript
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
